src/Grid/EventButton.vala

```diff
@@ -49,14 +50,31 @@ public class Maya.View.EventButton : Gtk.Revealer {
         event_box.add (internal_grid);
         event_box.button_press_event.connect ((event) => {
             if (event.type == Gdk.EventType.2BUTTON_PRESS && event.button == Gdk.BUTTON_PRIMARY) {
-                E.Source src = comp.get_data ("source");
-                if (src.writable == true && Model.CalendarModel.get_default ().calclient_is_readonly (src) == false) {
-                    edition_request ();
-                    return true;
+                edit_event ();
+            } else if (event.type == Gdk.EventType.BUTTON_PRESS && event.button == Gdk.BUTTON_SECONDARY) {
+                if (menu == null) {
+                    E.Source src = comp.get_data ("source");
+
+                    menu = new Gtk.Menu ();
+                    menu.attach_to_widget (this, null);
+                    var edit_item = new Gtk.MenuItem.with_label (_("Edit…"));
+                    var remove_item = new Gtk.MenuItem.with_label (_("Remove"));
+                    if (src.writable != true && Model.CalendarModel.get_default ().calclient_is_readonly (src) != false) {
```

Comment:
This test for writability is repeated in edit_event and remove event.  If it could change during the lifetime of the instance, it could be made a property e.g. 
```
private bool can_edit {
    get {
         ...
    }
}
```
Otherwise it could just be determined once on creation and assigned to a variable.